### PR TITLE
Added Deep Convolutional GAN

### PR DESCRIPTION
**Pull request by indiradutta**

### CODE REVIEW

**Comment by pranjaldatta** [comment]
Related to #31

File: pyvision/gans/deep_convolutional_gan/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+- torchvision==0.9.0
+- numpy==1.20.3
+- matplotlib==3.3.4
+- IPython==7.23.1
+- gdown==3.13.0
+
+## Dataset
+The original paper had used three datasets for training the DCGAN namely - *Large-scale Scene Understanding (LSUN) (Yu et al., 2015), Imagenet-1k and a newly assembled Faces dataset*. However due to computational and other limitations, we have used <a href = "http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html">Large-scale CelebFaces Attributes (CelebA) Dataset</a>.
+
+### Citation
```
Comment (pranjaldatta) [inline pyvision/gans/deep_convolutional_gan/README.md:17]: The citation comes in the end.

File: tests/gans/deep_convolutional_gan/gan_test.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from pyvision.gans.deep_convolutional_gan import Deep_Conv_GAN
```
Comment (pranjaldatta) [inline tests/gans/deep_convolutional_gan/gan_test.py:1]: Class names usually dont have underscores according to Python convention. Change it to `DeepConvGAN`

File: pyvision/gans/deep_convolutional_gan/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+```
+<br>
+
+**Note**: You may use any other dataset of your choice. However, please ensure that the directory structure remains the same for the code to be compatible with it.
+
+## Quick Start
+- Incase you want to use some other dataset to train the DCGAN, please initialize the DCGAN module with your desired dataset path and train as:
+
+```python
+from dcgan import DCGAN
```
Comment (pranjaldatta) [inline pyvision/gans/deep_convolutional_gan/README.md:53]: Seems like this documentation is inconsistent wrt to the test file provided wherein the import is from `pyvision.gans.deep_convolutional_gan` (i.e. repo root level) whereas here it seems like it is from the module level.

File: pyvision/gans/deep_convolutional_gan/README.md (hunk 1 above)
Comment (indiradutta) [reply pyvision/gans/deep_convolutional_gan/README.md]: okay, modified.

File: tests/gans/deep_convolutional_gan/gan_test.py (hunk 2 above)
Comment (indiradutta) [reply tests/gans/deep_convolutional_gan/gan_test.py]: okay, modified.

File: pyvision/gans/deep_convolutional_gan/README.md (hunk 3 above)
Comment (indiradutta) [reply pyvision/gans/deep_convolutional_gan/README.md:43]: For this one, we took reference from the pspnet module, can you elaborate a bit on the issue? I am not sure if I understand it clearly!

Comment (srijarkoroy) [reply pyvision/gans/deep_convolutional_gan/README.md:43]: We have made the required change now!

File: pyvision/gans/deep_convolutional_gan/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+```
+<br>
+
+**Note**: You may use any other dataset of your choice. However, please ensure that the directory structure remains the same for the code to be compatible with it.
+
+## Quick Start    
+- Incase you want to use some other dataset to train the DCGAN (with 1 GPU), please initialize the DCGAN module with your desired dataset path and train as:
+
+```python
+from pyvision.gans.deep_convolutional_gan.dcgan import DCGAN
```
Comment (pranjaldatta) [inline pyvision/gans/deep_convolutional_gan/README.md:43]: I really dont understand the structure. Why is `dcgan` a sub folder under deep_convolutional_gans?

**Review by pranjaldatta — COMMENTED** [review]
Merging for now since most are isolated changes. Would need fixing later.

**Review by pranjaldatta — APPROVED** [review]

File: pyvision/gans/deep_convolutional_gan/README.md (hunk 4 above)
Comment (indiradutta) [reply pyvision/gans/deep_convolutional_gan/README.md:43]: Um...deep_convolutional_gan is the main folder that has all the json files and all and dcgan is the folder for holding the DCGAN model that is the implementation

**Pull request merged**